### PR TITLE
chore: add linter as workflow

### PR DESCRIPTION
**Pull request by apeabody**
Test Example: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/actions/runs/4856014576/jobs/8655130539

This assumes all repos are using the same `./Makefile` format, otherwise falls back to devl-tools `1`.

Removes the current lint as a required check, but otherwise untouched to provide easy roll-back.

### CODE REVIEW

**Comment by apeabody** [comment]
@bharathkkb -  given the impact waiting on apply till after initial feedback

File: infra/terraform/test-org/github/protection.tf
Comment (bharathkkb) [inline infra/terraform/test-org/github/protection.tf:138]: It would be good to have an opt out flag to manage this in repo if any repos additional setup/params

File: infra/terraform/test-org/github/resources/lint.yaml
Comment (bharathkkb) [inline infra/terraform/test-org/github/resources/lint.yaml:22]: Could we add make this into a tf template and allow passing in env vars? I saw a few repos which pass in custom exclude dirs, disable TF lint etc

https://github.com/terraform-google-modules/terraform-example-foundation/blob/e6089046d93be6a397509ae63b40ff78037a645a/build/lint.cloudbuild.yaml#L21
https://github.com/terraform-google-modules/terraform-google-gcloud/blob/84847e60a758f3d490e84a3efcf20b7a6a73955e/build/lint.cloudbuild.yaml#L20

**Review by bharathkkb — APPROVED** [review]
Thanks @apeabody looks great!
Let's give a heads up to the maintainers on Monday before we apply so they are aware

File: infra/terraform/test-org/github/resources/lint.yaml
Comment (apeabody) [reply infra/terraform/test-org/github/resources/lint.yaml:22]: @bharathkkb - Yes, I thought about using it for the branch using data, but ran out of time last week.

I also noticed at least one repo doesn't put the `Makefile` in `.`: https://github.com/GoogleCloudPlatform/terraform-ecommerce-microservices-on-gke. My current thought is to simply fall back to a sensible default (e.g. `1` or `1.11`), but if needed we could plumb through a version override as well.

File: infra/terraform/test-org/github/protection.tf
Comment (apeabody) [reply infra/terraform/test-org/github/protection.tf:142]: add `disable_lint_yaml` to opt out of managed `lint.yaml`

File: infra/terraform/test-org/github/protection.tf
Comment (apeabody) [inline infra/terraform/test-org/github/protection.tf:183]: @bharathkkb - I'd rather leave this even if `disable_lint_yaml = true` as this is authoritative and a linter with the same name should be in-place.

File: infra/terraform/test-org/github/resources/lint.yaml
Comment (apeabody) [reply infra/terraform/test-org/github/resources/lint.yaml]: I've mapped through EXCLUDE_LINT_DIRS, set the branch based on the repo's default, and also if no `./Makefile` it will run with `1`.  Do we really want to provide an option to disable tf_lint?

File: infra/terraform/test-org/github/protection.tf
Comment (bharathkkb) [reply infra/terraform/test-org/github/protection.tf:183]: Yup sounds good

File: infra/terraform/test-org/github/resources/lint.yaml
Comment (bharathkkb) [reply infra/terraform/test-org/github/resources/lint.yaml]: @apeabody Could we check `./Makefile` and `infra/Makefile`. Between those two I think 99% of cases should be covered.

Comment (apeabody) [reply infra/terraform/test-org/github/resources/lint.yaml]: Sure, we could even do something like `$(find . -name Makefile -print -quit)`, the only risk is a repo with multiple `Makefile`, but none in the root, might match the wrong one.

**Comment by apeabody** [comment]
applied - lint / lint checks are n/a

**Pull request merged**
